### ISSUE TITLE
Missing step to add permissions

### ISSUE THREAD

**Issue by gewoonwoutje** [microsoftgraph/msgraph-training-nodeexpressapp#37]
I'm missing an action in step in `Register the app in the portal`. 

I got an error 'invalid_request'. I searched for permissions, and saw that only `user.read` was set by default. After adding the other three permissions, the app worked like expected and I received an access token.

![image](https://user-images.githubusercontent.com/9407072/86499983-64100a80-bd8e-11ea-8d28-f7ecb134c63a.png)

**Comment by jasonjoh** [comment]
You do not need to set these permissions explicitly on the app registration. The application requests them dynamically, by passing the scopes set in the .env file:

https://github.com/microsoftgraph/msgraph-training-nodeexpressapp/blob/5daa26525de01285f8b0a0449085be892976ebee/demo/graph-tutorial/.env.example#L4

https://github.com/microsoftgraph/msgraph-training-nodeexpressapp/blob/e09f5e9848145b4c59dae79cd8f455452ab6afe8/demo/graph-tutorial/app.js#L90

**Comment by gewoonwoutje** [comment]
Hm, that's weird. It didn't work before I set those permissions explicitly, it did work after...

**Comment by jasonjoh** [comment]
When you click **Sign in**, what's the URL that you're redirected to? Does it have the `v2.0` segment in the URL? It should look like:

```
https://login.microsoftonline.com/common/oauth2/v2.0/authorize?redirect_uri=http%3a%2f%2flocalhost%3a3000%2ffoo%2fauth%2fcallback
&response_type=code+id_token&response_mode=form_post
&client_id=765e28cf-bfab-4c23-965f-411e56ee004b
&prompt=login
&state=0kP7qCr1K2TUd4IrNUHHpkb1seOTQkbA
&nonce=UWKaG_jBjQQopPY7N8WYkmgascAdEQtv
&scope=profile+offline_access+user.read+calendars.read+openid
&x-client-SKU=passport-azure-ad
&x-client-Ver=4.0.0
&sso_nonce=AQABAAAAAAAGV_bv21oQQ4ROqh0_1-tA1A9au4V4ZsIsLrUFnIVzvEVg0fSiDL1GW7KwZPQHe_wN335Iwrz6UHTLyzrg7-iyKlJdG7Gt-nvSb_uX_UH2giAA
&client-request-id=024a895a-5e7a-427c-baea-82757d1c4437
&mscrid=024a895a-5e7a-427c-baea-82757d1c4437
```

The important bit being the `scope` parameter. By passing the list of scopes there, you dynamically request specific scopes.

**Comment by gewoonwoutje** [comment]
https://login.microsoftonline.com/common/oauth2/v2.0/authorize?redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Fcallback
&response_type=code%20id_token
&response_mode=form_post
&client_id=de938f05-faa0-425e-ad8d-f87d38602945
&prompt=login
&state=4fkxtyv9eCVfPra7K8udur2ca9RfLQ5l
&nonce=QMJyNShmuupaqhYjFScqvi557XFxL8t5
&scope=profile%20offline_access%20user.read%20calendars.read%20openid
&x-client-SKU=passport-azure-ad
&x-client-Ver=4.0.0

It has %20 separators in the scope.

**Comment by jasonjoh** [comment]
That shouldn't matter, that all looks fine. Out of curiosity, what line of code gave invalid request before you changed your app registration?

**Comment by gewoonwoutje** [comment]
I'm afraid I don't know. I saw it as an error message in the UI (I believe a flash error message?). I always find it hard to find where errors originate in that express middleware magic. The change in the permissions made sense to me, and it seemed to work out. I figured it was something missing in the manual, but that was wrong I guess.

Removing the permissions does not make the error return, so I guess I changed something else as well, hard to remember what exactly 😅.
